python/pyspark/sql/functions.py

```diff
@@ -2200,38 +2200,70 @@ def make_date(year: "ColumnOrName", month: "ColumnOrName", day: "ColumnOrName")
     return Column(jc)
 
 
+@overload
+def date_add(start: "ColumnOrName", days: "ColumnOrName") -> Column:
+    ...
+
+
+@overload
 def date_add(start: "ColumnOrName", days: int) -> Column:
+    ...
```

Comment:
We don't really need overloads here ‒ these are useful when there are some relationships between arguments and / or return type.

Reply:
ack- will remove, thank you! I thought it would be a good idea for consistency with the other functions, but you're right in that date_add (+ others I've modified) will output the same thing regardless